src/Compilers/Core/Portable/Emit/EditAndContinue/DeltaMetadataWriter.cs

```diff
@@ -1461,6 +1535,8 @@ private bool TryGetExistingMethodImplIndex(MethodImplKey item, out int index)
 
         private sealed class ParameterDefinitionIndex : DefinitionIndexBase<IParameterDefinition>
         {
+            private int _addedInPreviousGenerations;
```

Comment:
>_addedInPreviousGenerations [](http://example.com/codeflow?start=24&length=27)

Does `DefinitionIndex<T>` handle cases where the definition map is a collection of existing and new definitions? In short, would it make sense to replace `ParameterDefinitionIndex` with `DefinitionIndex<IParameterDefinition>` with an appropriate `TryGetExistingParameterMapIndex` helper method?

Reply:
I've done this in [`4362ce0` (#54835)](https://github.com/dotnet/roslyn/pull/54835/commits/4362ce0339d01d9870f563cca8106ac10245f694)

It allowed me to remove `ParameterDefinitionIndex` so its a bit neater, but it does mean adding updated parameters to two different collections, once because we need to know the row number for emitting custom attributes, and one because we need to iterate through all parameters whether updated or not. Strictly I suspect that last part is unnecessary for just custom attributes, but it will be needed for parameter renames anyway.